uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/ui/screens/habits/show/views/HistoryCard.kt

```diff
@@ -153,10 +153,15 @@ class HistoryCardPresenter(
             val series = if (habit.isNumerical) {
                 entries.map {
                     when {
-                        it.value == Entry.UNKNOWN -> OFF
-                        (habit.targetType == AT_MOST) && (it.value / 1000.0 <= habit.targetValue) -> ON
-                        (habit.targetType == AT_LEAST) && (it.value / 1000.0 >= habit.targetValue) -> ON
-                        else -> GREY
+                        it.value == SKIP -> {
+                            HistoryChart.Square.HATCHED
+                        }
+                        it.value > 0 -> {
```

Comment:
As mentioned before, better to have an exhaustive match (every single value) rather than whole range (`>0`).